pkg/plaid_cli/linker.go

```diff
@@ -5,44 +5,75 @@ import (
 	"fmt"
 	"log"
 	"net/http"
+	"os"
 	"text/template"
 
 	"github.com/plaid/plaid-go/plaid"
 	"github.com/skratchdot/open-golang/open"
 )
 
 type Linker struct {
-	Results    chan string
-	Errors     chan error
-	Client     *plaid.Client
-	ClientOpts plaid.ClientOptions
-	Data       *Data
+	Results       chan string
+	RelinkResults chan bool
+	Errors        chan error
+	Client        *plaid.Client
+	ClientOpts    plaid.ClientOptions
+	Data          *Data
 }
 
 type TokenPair struct {
 	ItemID      string
 	AccessToken string
 }
 
-func (l *Linker) Relink(itemID string, port string) (*TokenPair, error) {
+func (l *Linker) Relink(itemID string, port string) error {
 	token := l.Data.Tokens[itemID]
-	res, err := l.Client.CreatePublicToken(token)
+	hostname, err := os.Hostname()
 	if err != nil {
-		return nil, err
+		log.Fatal(err)
 	}
-
-	return l.link(port, handleRelink(l, res.PublicToken))
+	client, err := plaid.NewClient(l.ClientOpts)
+	resp, err := client.CreateLinkToken(plaid.LinkTokenConfigs{
+		User: &plaid.LinkTokenUser{
+			ClientUserID: hostname,
+		},
+		ClientName:   "plaid-cli",
+		CountryCodes: []string{"US"},
+		Language:     "en",
+		AccessToken:  token,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	return l.relink(port, resp.LinkToken)
 }
 
 func (l *Linker) Link(port string) (*TokenPair, error) {
-	return l.link(port, handleLink(l))
+	hostname, err := os.Hostname()
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err := plaid.NewClient(l.ClientOpts)
+	resp, err := client.CreateLinkToken(plaid.LinkTokenConfigs{
+		User: &plaid.LinkTokenUser{
+			ClientUserID: hostname,
+		},
+		ClientName:   "plaid-cli",
+		Products:     []string{"transactions"},
```

Comment:
Does `plaid-cli` require access to any of the other Plaid products? Options are `transactions, auth, identity, assets, investments, liabilities, payment_initiation`.